crates/codegen/parser_runtime/src/templates/language.tera

```diff
@@ -0,0 +1,236 @@
+#[cfg(feature = "slang_napi_interfaces")]
+use { napi::bindgen_prelude::*, napi_derive::napi };
+
+use semver::Version;
+
+use super::{
+    cst,
+    kinds::{RuleKind, TokenKind, ProductionKind},
+    parse_output::ParseOutput,
+    support::*,
+};
+
+#[cfg( feature = "slang_napi_interfaces" )]
+use super::napi::napi_parse_output::ParseOutput as NAPIParseOutput;
+
+#[derive(Debug)]
+#[cfg_attr( feature = "slang_napi_interfaces", napi(namespace = "language") )]
+pub struct Language {
+    pub(crate) version: Version,
+    {%- for version in code.referenced_versions -%}
+        pub(crate) version_is_at_least_{{ version | replace(from=".", to="_") }}: bool,
+    {%- endfor -%}
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Unsupported {{ language_name }} language version '{0}'.")]
+    UnsupportedLanguageVersion(Version),
+
+    #[cfg( feature = "slang_napi_interfaces" )]
+    #[error("Invalid semantic version '{0}'.")]
+    InvalidSemanticVersion(String),
+}
+
+#[cfg( feature = "slang_napi_interfaces" )]
+impl From<Error> for napi::Error {
+    fn from(value: Error) -> Self {
+        napi::Error::from_reason(value.to_string())
+    }
+}
+
+impl Language {
+    const VERSIONS: &'static [&'static str] = &[
+        {%- for version in versions -%}
+            "{{ version }}",
+        {%- endfor -%}
+    ];
+
+    pub fn new(version: Version) -> std::result::Result<Self, Error> {
+        if Self::VERSIONS.contains(&version.to_string().as_str()) {
+            Ok(Self {
+                {%- for version in code.referenced_versions -%}
+                    {# Separator #}
+                    version_is_at_least_{{ version | replace(from=".", to="_") }}: Version::parse("{{ version }}").unwrap() <= version,
```

Comment:
I wonder if we can use the newly added filter?

```suggestion
                    version_is_at_least_{{ version | snake_case }}: Version::parse("{{ version }}").unwrap() <= version,
```

Reply:
Hmm. it probably would work. Didn't think of it - the version string isn't really in the expected domain.